File: Programmers/Lv.1/정수 제곱근 판별.java
/**
 * fileName : 정수 제곱근 판별.java
 * author :  한찬영
 * date : 2023-03-07
 * description :
 * ===========================================================
 * DATE                 AUTHOR                NOTE
 * -----------------------------------------------------------
 * 2023-03-07            한찬영             최초 생성
 */

import java.lang.Math;

class Solution {
    public long solution(long n) {
        long answer = 0;
        
        double sqr = Math.sqrt(n);
       
        
        if(sqr ==(long)sqr){
            answer = ((long)sqr + 1)*((long)sqr + 1);
        }
        else{
            answer = -1;
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/정수 제곱근 판별.py
/**
 * fileName : 정수 제곱근 판별.py
 * author :  한찬영
 * date : 2023-04-04
 * description :
 * ===========================================================
 * DATE                 AUTHOR                NOTE
 * -----------------------------------------------------------
 * 2023-04-04            한찬영             최초 생성
 */

import math
def solution(n):
    answer = 0
    
    sqr_n = math.sqrt(n)
    sqr_n1 = sqr_n *10
    
    if(sqr_n1%10 == 0):
        answer = math.pow(sqr_n+1,2)
    else:
        answer = -1
    
    
    return answer
